src/components/profile-view/profile-view.jsx

```diff
@@ -0,0 +1,115 @@
+import {useState} from "react";
+
+import {UserInfo} from "./user-info";
+import {FavoriteMovies} from "./favorite-movies";
+import {UpdateUser} from "./update-user";
+import {Row, Container, Col} from "react-bootstrap";
+
+export const ProfileView = ({
+  user,
+  setUser,
+  token,
+  movies,
+  removeFavorite,
+  movie,
+}) => {
+  const [username, setUsername] = useState(user.Username);
+  const [password, setPassword] = useState(user.Password);
+  const [email, setEmail] = useState(user.Email);
+  const [birthday, setBirthday] = useState(user.Birthday);
+
+  const handleSubmitUpdate = (event) => {
+    event.preventDefault();
+
+    const data = {
+      Username: username,
+      Password: password,
+      Email: email,
+      Birthday: birthday,
+    };
+
+    fetch(
+      `https://myflix-database-api-9ba401fe0e70.herokuapp.com/users/${user.Username}`,
+      {
+        method: "PUT",
+        body: JSON.stringify(data),
+        headers: {
+          "Content-Type": "application/json",
+          Authorization: `Bearer ${token}`,
+        },
+      }
+    )
+      .then((response) => {
+        if (response.ok) {
+          alert("User info update successful");
+          window.location.reload();
+        } else {
+          alert("User info update failed");
+        }
+      })
+      .then((user) => {
+        localStorage.setItem("user", JSON.stringify(data));
+        setUser(user);
+      });
+  };
+
+  const handleSubmitDeregister = (event,  onLoggedOut) => {
+    event.preventDefault();
+
+    const data = {
+      Username: username,
+      Password: password,
+      Email: email,
+      Birthday: birthday,
+    };
+
+    fetch(
+      `https://myflix-database-api-9ba401fe0e70.herokuapp.com/users/${user.Username}`,
+      {
+        method: "DELETE",
+        body: JSON.stringify(data),
+        headers: {
+          "Content-Type": "application/json",
+          Authorization: `Bearer ${token}`,
+        },
+      }
+    ).then((response) => {
+      if (response.ok) {
+        alert ("Deregister succesful");
+        window.location.replace("/signup");
+        onLoggedOut();
+      } else {
+        alert("Deregister failed");
+      }
+    });
+
+ 
+
+  return (
+    <Container>
+      <Row>
+        <Col xs={12} sm={6}>
+          {" "}
+          <UserInfo
+            name={user.Username}
+            email={user.Email}
+            handleSubmitDeregister={handleSubmitDeregister}
```

Comment:
1. If you check your API, you'll notice that the endpoint for removing a user doesn't require a payload in the request body.
2. The function `handleSubmitDeregister` was defined with two parameters but was called without any argument(implicitly with the event argument). Regardless, that function should only call the remove user endpoint then call the `onLoggout` function defined in the main-view(you can pass it as a prop). Then you can safely redirect to the signup path.

Reply:
1. so i just remove: body: JSON.stringify(data)?
2. do i call the function here:
`    <UserInfo
            name={user.Username}
            email={user.Email}
            handleSubmitDeregister={handleSubmitDeregister(and which argument do i need to put here?);}`
4. how do i call the remove user endpoint? 
5.  i already passed the onLoggedOut function as prop, does it need to be called with an argument aswell?

Reply:
Ok, so i changed it to now:
`
    fetch(
      `https://myflix-database-api-9ba401fe0e70.herokuapp.com/users/${user.Username}`,
      {
        method: "DELETE",
        headers: {
          "Content-Type": "application/json",
          Authorization: `Bearer ${token}`,
        },
      }
    ).then((response) => {
      if (response.ok) {
        onLoggedOut();
        window.location.replace("/signup");
      } else {
        alert("Deregister failed");
      }
    });
  };`

and i pass the onLoggedOut={handleLogout} as props from main-view to profile-view. but it gives me onLoggedOut is not a function. Thats why i need to give it an argument right? 

I defined the function in the main-view like this
`const handleLogout = () => {
    setUser(null);
    setToken(null);
    localStorage.clear();
  };`

Which arguments do i need to call it with?